lms/djangoapps/courseware/views/views.py

```diff
@@ -737,6 +737,13 @@ def _progress(request, course_key, student_id):
         and certs_api.cert_generation_enabled(course_key)
     )
 
+    # If current certificate is invalidated by instructor
+    # then don't show the generate button.
+    is_invalid = False
+    if show_generate_cert_btn and certs_api.is_certificate_invalid(student, course_key):
```

Comment:
After a discussion with @cpennington, we agree that Performance out-weights readability of the code here. I agree we can leave this as it is here. 
However, I created ECOM-4681 for tech debt to clean this set of logic up better.